app/src/main/java/org/wikipedia/search/RecentSearchesFragment.kt

```diff
@@ -62,12 +67,19 @@ class RecentSearchesFragment : Fragment() {
         binding.addLanguagesButton.setOnClickListener { onAddLangButtonClick() }
         binding.recentSearchesRecycler.layoutManager = LinearLayoutManager(requireActivity())
         binding.namespacesRecycler.layoutManager = LinearLayoutManager(requireActivity(), RecyclerView.HORIZONTAL, false)
-        binding.recentSearchesRecycler.adapter = RecentSearchAdapter()
+
+        suggestedSearchTerm = requireActivity().intent.getStringExtra(SearchActivity.EXTRA_SUGGESTED_QUERY)
+        binding.recentSearchesRecycler.adapter = if (suggestedSearchTerm.isNullOrEmpty()) RecentSearchAdapter() else ConcatAdapter(SuggestedSearchAdapter(), RecentSearchAdapter())
+        binding.recentSearchesHeaderContainer.isVisible = suggestedSearchTerm.isNullOrEmpty()
+
         val touchCallback = SwipeableItemTouchHelperCallback(requireContext())
         touchCallback.swipeableEnabled = true
         val itemTouchHelper = ItemTouchHelper(touchCallback)
         itemTouchHelper.attachToRecyclerView(binding.recentSearchesRecycler)
         setButtonTooltip(binding.recentSearchesDeleteButton)
+
+        RabbitHolesEvent.submit("impression", "search")
```

Comment:
Do we want to check whether the rabbit hole is enabled?

Reply:
Whoops, I see what you mean; I've updated the RabbitHolesEvent send itself only if the experiment is active.